nixos/modules/services/networking/hostapd.nix

```diff
@@ -211,7 +213,7 @@ in
         wantedBy = [ "multi-user.target" ];
 
         serviceConfig =
-          { ExecStart = "${pkgs.hostapd}/bin/hostapd ${configFile}";
+          { ExecStart = "${pkgs.hostapd}/bin/hostapd /etc/hostapd/hostapd.conf";
```

Comment:
this also requires a `restartTriggers` entry for `configFile`, otherwise hostpad will not be restarted when the config file changes.

not very enthused by making stuff globally accessible though to be honest. would be much nicer if there was some kind of (interactive?) nix-diff that takes two derivations and a path, then diffs the closures of that path in both inputs.

Reply:
> this also requires a restartTriggers entry for configFile, otherwise hostpad will not be restarted when the config file changes.

Thanks I will update it.

> not very enthused by making stuff globally accessible though to be honest.

I have seen this trend in other packages as well and thought it could be useful for `hostapd` as well. Is there any downside in making them globally accessible?

> would be much nicer if there was some kind of (interactive?) nix-diff that takes two derivations and a path, then diffs the closures of that path in both inputs.

I tried `nix-diff` and it works okay, but of course it is easier to just use `diff -u` on the resulting builds.

I am not too invested in this, but I think that all packages should follow the same guideline, whatever that may be.

Reply:
exposing things in `/etc` that needn't be there for technical reasons (eg because a client needs access to a shared config) can introduce non-reproducibility and other odd effects in ways that are not obvious. imagine a service that can optionally read the hostapd config file when it's present and act on its contents. if the config is exposed in `/etc` that service would not be notified of changes on the nixos level, which can in turn cause very strange behaviour. it's usually best to expose as little in `/etc` as possible to avoid the possibility of these things entirely. a guideline in the manual that states exactly what is and isn't a good thing to put in `/etc` would certainly be helpful.

that's not to say that being able to easily diff configs isn't a useful thing to do! just that exposing it may not be the best way to do it. (nix-diff itself also isn't a terribly great solution on its own, but it could be a good start. maybe a wrapper script that takes two derivations and a file path, diffs the file path in both, and recursively extracts paths for further diffing? :thinking:)

Reply:
> imagine a service that can optionally read the hostapd config file when it's present and act on its contents.

Currently, such a service would not even be possible to implement because the path to the generated config file is hidden in the `let` statement in the module. So the choice is between not being able to implement such a service at all, or with reproducibility bugs.

Also not all access to such a globally exposed file can be imagined via services that rely on being notified about changes. Other access can be done by one-off programs/scripts. For example, lets say you have a number of hosts where you want to get the wpa password from, if that host happens to run `hostapd`.

```sh
for host in hostA hostB hostC; do
   ssh $host grep wpa_passphrase= /etc/hostapd/hostapd.conf
done
```

Currently, such a simple script would not be possible without doing some weird parsing of the generated systemd service files.


I want to give some time for other opinions to be voiced and then I can close this PR.

Reply:
> not very enthused by making stuff globally accessible

👍

There is an unofficial unwritten "policy" to support what @pennae says.